fair_svm/fair_svm.py

```diff
@@ -0,0 +1,356 @@
+""" A skikit-learn compatible, provably fair binary classifier using logistic regression """
+
+from __future__ import division
+import numpy as np
+from sklearn.base import BaseEstimator, ClassifierMixin
+from sklearn.utils.validation import check_X_y, check_array, check_is_fitted
+from sklearn.utils.extmath import log_logistic, safe_sparse_dot
+from sklearn.utils.fixes import expit
+from sklearn.preprocessing import OneHotEncoder
+from scipy.optimize import minimize # for loss func minimization
+
+
+from sklearn.metrics.pairwise import linear_kernel, polynomial_kernel, rbf_kernel, sigmoid_kernel
+
+
+def _svm_dual(a, gram_matrix, y):
```

Comment:
Are there scikit-learn implementations of this and `_svm_dual_jac` that we could use? Not that there's anything wrong with these, but we could at least compare.